nautobot/docs/user-guide/administration/installation/health-checks.md

```diff
@@ -0,0 +1,307 @@
+# Nautobot Health Checks
+
+In a production deployment of Nautobot, you'll want health checks (also termed liveness checks, readiness checks, etc.) for each distinct component of the Nautobot system, so as to be able to detect if any component fails and ideally respond automatically. While this topic can be (and is) the subject of multiple books, this document attempts to provide some basic "best practices" guidelines. If you're deploying Nautobot as part of a larger enterprise system, of course you'll want to follow your organization's experts and their guidance, but if you're "on your own", you can do worse than starting here.
+
+## Health Check Approaches
+
+In general the following commands or HTTP requests can serve as health checks for the various components of a Nautobot system.
+
+### Nautobot HTTP Server
+
+In addition to simply monitoring the existence of the `nautobot-server` process ID, two more in-depth approaches are possible here.
+
+An HTTP `GET` request to `<server>/health/` should return an HTTP `200 OK` response so long as:
+
+- the server is running
+- and the server can connect to the database
+- and all Django migrations have been applied
+- and the server can connect to Redis
+- and the server can write to an appropriate location on the filesystem
+- and the server is not too busy handling other requests to respond to this request.
+
+Similarly, but not identically, the CLI command `nautobot-server health_check` should run and return an exit code of `0` (success) so long as:
+
+- the command can connect to the database
+- and all Django migrations have been applied
+- and the command can connect to Redis
+- and the command can write to an appropriate location on the filesystem
+
+Note the differences between these two. In some situations you'll want to use both for different types of checks. More on this later in this document.
+
+### Nautobot Celery Worker
+
+In addition to monitoring the existence of a given Celery worker process ID, you can use the fact that Celery provides a [`celery inspect ping` CLI command](https://docs.celeryq.dev/en/stable/reference/cli.html#celery-inspect) that sends a short message to a given Celery worker(s) and reports back on whether it receives a response(s). Nautobot wraps this with the `nautobot-server` CLI command, so in general you can run `nautobot-server celery inspect ping --destination <worker name>` to confirm whether a given worker is able to receive and respond to Celery control messages.
```

Comment:
Mention how the worker name is derived from the server hostname unless overridden with `nautobot-server celery worker -n <new worker name>`?